odoo/fields.py

```diff
@@ -38,12 +39,29 @@
 )
 from .tools.sql import pg_varchar
 from .tools.mimetypes import guess_mimetype
-from .tools.misc import unquote, has_list_types
+from .tools.misc import unquote, has_list_types, DefaultType, DEFAULT
 from .tools.translate import html_translate, _
 
 from odoo.exceptions import CacheMiss
 from odoo.osv import expression
 
+import typing
+if typing.TYPE_CHECKING:
+    try:
+        from typing_extensions import Self
+    except ImportError:
+        from typing import Self
+    from odoo.models import BaseModel
+    from collections.abc import Mapping
+    M = typing.TypeVar("M", bound=BaseModel)
+    ModelId = int | NewId
+    TypeDomain = list[str | tuple[str, str, typing.Any]]
+    TypeContext = Mapping[str, typing.Any]
```

Comment:
Shouldn't those be defined in `models.py` and imported here?

Reply:
I think `api.py` is a better place, that is where the environment is defined and context is related to the environment.